### ISSUE TITLE
vue-loader not processing images in dynamic component

### ISSUE THREAD

**Issue by jh-thank-you** [vuejs/vue-loader#814]
Vue Team,

Thanks for all that you do for the dev community.

**I'm not sure if this is an issue or a configuration problem.** One of the contributors to webpack has been helping me sort an issue but we are at a point where he thinks it may be a vue-loader issue.

I have a dynamic component/slot that is called via the data captured and passed by an Event Bus. This all work great and in development mode everything works as expected. When I build out the production folder the dynamic .vue template files are processed and placed in the correct directory (in the assets/modals folder) but none of the hard/static coded src links or src images in the template files are processed (no image hash and link hash updates are applied).

I have a setup on github that's working via a hack (I am using Copy Webpack Plugin to copy over all the files in the directory) but this is just a quick fix for the above outlined issue.

Here is the github repository:
https://github.com/jh-thank-you/webpack-w-gsap

Thanks again.

All the best,

Jim

**Comment by LinusBorg** [comment]
Can you tell me a specific component that this is happening for so I don't havd to look through your whole project? 

That would be helpful, thanks :)

**Comment by jh-thank-you** [comment]
Thank you (very much) for the help.

Yes, in the assets/modals folder any of the three modals ( for example modalPrintAdvil.vue) have src files (which are hard coded/not dynamic links) that are not being processed and hashed like the other non-dynamic components.

I hope clears things up. 

Thanks again.

**Comment by yyx990803** [comment]
You need to use relative paths (start with `./`) or start the url with `~` for it to be considered a module request.

**Comment by jh-thank-you** [comment]
Evan,

Thank you for taking the time to look into this.

My webpack aliases are set to` ./ `and `~` — alias (shortened) paths only work in CSS files with `~`
```    
      alias: { 
      'vue$': 'vue/dist/vue.esm.js',
      assets: path.resolve(__dirname, './assets'),
      components: path.resolve(__dirname, './assets/components'),
      css: path.resolve(__dirname, './assets/css'),
      fonts: path.resolve(__dirname, './assets/fonts'),
      img: path.resolve(__dirname, './assets/img'),
      js: path.resolve(__dirname, './assets/js'),
      modals: path.resolve(__dirname, './assets/modals'),
      scss: path.resolve(__dirname, './assets/scss'),
    },
```
In the modalPrintAdvil.vue template component - I'm not sure why (for the production build):

This shorted path alias does not work:
`src="~assets/modalPrintAdvil1c.jpg"`

This relative path does not work:
`src="./assets/img/modalPrintAdvil1c.jpg"`

**This works:**
`src="~assets/img/modalPrintAdvil1c.jpg"`

I will also share this with Juho, one of the webpack contributors, and see if he can shine a bit of light on why the path aliases within a vue template component are not working. I will report back.

Also, I forgot to point out, the modal-button.vue component/template file has a programmatic link
`:src="'assets/img/' + this.id + '1a-sm.png'"`

This works great for the dev build but for production it does not process the images.  As a workaround I have placed all of these thumbnail images in `assets/img-dynamic` and use copy-webpack-plugin  to copy over the files/directory. 

```
    new CopyWebpackPlugin([
        { from: 'assets/img-dynamic', to: 'assets/img-dynamic/[name].[ext]' }
    ]),
```
This partially solves the issue but it does not take advantage of hashed image names and links for caching.

I tried the `./` and `~` options with this modal-button component but none of these options worked.  

I have updated the repo:
https://github.com/jh-thank-you/webpack-w-gsap

Thanks again.

**Comment by nagisa1993** [comment]
Try this link 👉🏻[this](https://github.com/vuejs-templates/webpack/issues/450) works for me.

**Comment by kaankucukx** [comment]
>  `<img :src="require(``@/assets/${posts.img}``)" alt="">
> `
> 
> Care for back ticks , must have only 1.

> This is it.

**Comment by VladTzaru** [comment]
@kaankucukx Works, thanks a lot.

**Comment by Akronae** [comment]
@kaankucukx You're a life saver, thanks 😂

**Comment by viandmarket25** [comment]
I also had this problem almost stole my 48 hours,

Initially i thought the error was programatic, and then after googling researching i never even found out the cause of the error, At some point i read a post and the guy revealed that his missing assets were not bundled, Oh mY! so how do i solve this problem? the bundler is missing the files? what do i do? 

Initial tests:

1. new  URL('/src/assets/icons/'+name, import.meta.url), But this didn't save me as the function was not defined via browser access
2. Many others, including modifying my code.


Solution:
After thinking, I realized the solution was simple, just add the assets on the server and link to the url like you would reference an external url,

so on same server i created another folder "dynamic-assets"

The trick however is to know exactly your server root, and point your dynamic files directly,

Make sure your path starts with your server root eg: www.solve.com, concatenate it with your file path,
then serve the url string,

Next put your unhashed images in "dynamic-assets" and add them in your lib folder, so you have 
const imgUrl = 'www.solve.com/dynamic-assets/my-img.png'

Solved. 

Try find other solutions that you might prefer, cheers.
